### ISSUE TITLE
WSL [--set-version] fails with message 'Element not found' when converting to WSL 2.

### ISSUE THREAD

**Issue by martijnvanschie** [microsoft/WSL#5631]
# Environment

```none
Windows build number: [run `[Environment]::OSVersion` for powershell, or `ver` for cmd]
Your Distribution version: [On Debian or Ubuntu run `lsb_release -r` in WSL]
Whether the issue is on WSL 2 and/or WSL 1: [run `cat /proc/version` in WSL]
```
Windows build number:
Win32NT 10.0.19041.0 Microsoft Windows NT 10.0.19041.0

Current WSL1 Distro:
Release:        20.04

WSL Distro version:
Linux version 4.4.0-19041-Microsoft (Microsoft@Microsoft.com) (gcc version 5.4.0 (GCC) ) #1-Microsoft Fri Dec 06 14:06:00 PST 2019

# Steps to reproduce
After performing the WSL 2 upgrade using the guide:
https://docs.microsoft.com/en-us/windows/wsl/install-win10

I checked my current distributions.

```
❯ wsl -l -v
  NAME      STATE           VERSION
* Ubuntu    Running         1
```

Then i ran the update command to set the current Ubuntu distro to WSL 2.

```
> wsl --set-version ubuntu 2
Conversion in progress, this may take a few minutes...
For information on key differences with WSL 2 please visit https://aka.ms/wsl2
Element not found.
```
Then it fails with the message.

#  Expected behavior

The upgrade should succeed and the Distibution should be converted to WSL 2.

# Actual behavior

The upgrade fails and the Distibution stays WSL

# Log files
I've provided the etl trace files.
[set-version etl logs.zip](https://github.com/microsoft/WSL/files/4959420/set-version.etl.logs.zip)

**Comment by Lazare-C** [comment]
Same problem and I cant launch any wsl
![image](https://user-images.githubusercontent.com/18364305/88169291-144d9200-cc1c-11ea-9b99-538af7ab3ea7.png)

**Comment by ulziibat-n** [comment]
![image](https://user-images.githubusercontent.com/42606441/88209620-89a17d00-cc85-11ea-8dd4-dbed105ad884.png)

Same here ... 😭😭😭 What i should do?

**Comment by licanhua** [comment]
> Same here ... 😭😭😭 What i should do?

I didn't see any error on your picture:`wsl --set-default-version 2` looks good

**Comment by licanhua** [comment]
@benhillis  CreateNetworkEnd fails with 0x80070490

**Comment by benhillis** [comment]
Interesting.  Could somebody who is hitting this issue please take [networking logs](https://github.com/microsoft/WSL/blob/master/CONTRIBUTING.md#9-networking-logs)?

Thanks!

**Comment by martijnvanschie** [comment]
> ![image](https://user-images.githubusercontent.com/42606441/88209620-89a17d00-cc85-11ea-8dd4-dbed105ad884.png)
> 
> Same here ... 😭😭😭 What i should do?

I think yours works 👍 
Do a `wsl -l -v` to see the versions

**Comment by ulziibat-n** [comment]
Yes. Now works fine. 
After updated Windows 10 and WSL. My distro still works on WSL1. 
I'm uninstalled my distro, then installed back. After this quick step my distro works on WSL2.
Thanks all ...

**Comment by martijnvanschie** [comment]
> Interesting. Could somebody who is hitting this issue please take [networking logs](https://github.com/microsoft/WSL/blob/master/CONTRIBUTING.md#9-networking-logs)?
> 
> Thanks!

Hi @benhillis. I took a network log as described.
https://aka.ms/AA9286r

Hope this helps solve the issue.

Regards.

**Comment by imsumitrai** [comment]
Mine was working fine after restart it stop working.
I reinstalled package from https://aka.ms/wsl2-install
What to do?
![image](https://user-images.githubusercontent.com/16753136/88456580-2fc6d000-ce9c-11ea-9003-842a1df3bcd9.png)

**Comment by martijnvanschie** [comment]
Hi @imsumitrai. You have the same error message as I get.
I posted a network trace on the Feedback Hub so hopefully they can find something soon.

**Comment by martijnvanschie** [comment]
> Interesting. Could somebody who is hitting this issue please take [networking logs](https://github.com/microsoft/WSL/blob/master/CONTRIBUTING.md#9-networking-logs)?
> 
> Thanks!

Hi @benhillis . Did you get a chance to check the network log?
I'm really eager to get this problem solved.

Regards,
Martijn

**Comment by imsumitrai** [comment]
Hi @benhillis.
Please address this issue.
We all are stuck on this. Nothing works.
I am eagerly waiting for this.

Unfortunately my DockerDesktop stops working after this issues. It requires WSL2 working.
Unfortunately my docker desktop even won't able to run on Hyper-V enabled.

**Comment by imsumitrai** [comment]
If anyone found the solution please help me.
This error is killing me.

**Comment by gitsoup** [comment]
I got frustrated and reset the PC and its working. i am not sure how many
of you got the luxury of going thru that painful process of resetting. No
one there to help me, so I go through that self-sucidal approach. But its
working.

**Comment by OneBlue** [comment]
We acknowledge the issue and we are actively working on it. I'll update this thread when we'll have more information.

**Comment by martijnvanschie** [comment]
Good to hear its been picked up. Hope a solution is found soon 😀

**Comment by ootjersb** [comment]
I was able to overcome this error by uninstalling any network component (such as VPN client). I could also see within Hyper-V Manager whenever I created a new virtual machine that opening network settings was taking a long time, and after that time nothing appeared. Maybe your situation is similar.

**Comment by martijnvanschie** [comment]
The Hyper-V issues was also the case before i moved to WSL2 which does not require Hyper-V anymore. But i had the same "unresponsiveness" with the network settings. This is also trying to get the Virtual Network Adapters.

**Comment by imsumitrai** [comment]
Finally it works..

I performed network reset. after that my internet stopped working.
So I was able to figure out the culprit driver which is blocking my network.

After that I uninstalled that culprit driver and after a restart. Wewww everything start working.

For reference my adaptor configuration looks like this now.
If you have any issue either disable/ or uninstall other driver which you think is culprit and restart.
![image](https://user-images.githubusercontent.com/16753136/91536808-eb16d480-e932-11ea-93d2-1a791b415068.png)
![image](https://user-images.githubusercontent.com/16753136/91536835-fb2eb400-e932-11ea-97fd-2c3954f55889.png)
